iothub/device/iot-device-samples/pnp-device-sample/temperature-controller-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/TemperatureController.java

```diff
@@ -250,41 +229,52 @@ private static void initializeAndProvisionDevice() throws ProvisioningDeviceClie
     {
         SecurityProviderSymmetricKey securityClientSymmetricKey = new SecurityProviderSymmetricKey(deviceSymmetricKey.getBytes(StandardCharsets.UTF_8), registrationId);
         ProvisioningDeviceClient provisioningDeviceClient;
-        ProvisioningStatus provisioningStatus = new ProvisioningStatus();
 
         provisioningDeviceClient = ProvisioningDeviceClient.create(globalEndpoint, scopeId, provisioningProtocol, securityClientSymmetricKey);
 
         AdditionalData additionalData = new AdditionalData();
         additionalData.setProvisioningPayload(com.microsoft.azure.sdk.iot.provisioning.device.plugandplay.PnpHelper.createDpsPayload(MODEL_ID));
 
-        provisioningDeviceClient.registerDevice(new ProvisioningDeviceClientRegistrationCallbackImpl(), provisioningStatus, additionalData);
+        final CountDownLatch registrationLatch = new CountDownLatch(1);
+        AtomicReference<ProvisioningDeviceClientRegistrationResult> registrationResultReference = new AtomicReference<>();
+        AtomicReference<Exception> registrationExceptionReference = new AtomicReference<>();
+        provisioningDeviceClient.registerDevice(
+            (provisioningDeviceClientRegistrationResult, e, context) ->
+            {
+                registrationResultReference.set(provisioningDeviceClientRegistrationResult);
+                registrationExceptionReference.set(e);
+                registrationLatch.countDown();
+            },
+            null,
+            additionalData);
 
-        while (provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningDeviceClientStatus() != ProvisioningDeviceClientStatus.PROVISIONING_DEVICE_STATUS_ASSIGNED)
+        System.out.println("Waiting for Provisioning Service to register");
+        boolean timedOut = !registrationLatch.await(1, TimeUnit.MINUTES);
```

Comment:
Is this timeout a best practice approach?
Is there any documentation or additional comments we can provide about the choice of this 'timeout'?